### ISSUE TITLE
Incorrect probability density function for Gumbel

### ISSUE THREAD

**Issue by sidhantnagpal** [sympy/sympy#16521]
The formula for density (pdf) of [Gumbel distribution](https://en.wikipedia.org/wiki/Gumbel_distribution) is incorrect.
```py
In [2]: mu = Symbol("mu")                                                       

In [3]: beta = Symbol("beta", positive=True)                                    

In [4]: X = Gumbel("X", beta, mu) 

In [5]: density(X)(x)                                                           
Out[5]: 
  -(-μ + x)          
  ──────────         
      β        -μ + x
 ℯ           - ──────
                 β   
ℯ                    
─────────────────────
          β          
```
It was introduced in #11751. It should be:
```py
    -(-μ + x)          
    ──────────         
        β        -μ + x
 - ℯ           - ──────
                   β   
ℯ                      
───────────────────────
           β           

```
The [docstring](https://github.com/sympy/sympy/blob/master/sympy/stats/crv_types.py#L1459-L1497) for Gumbel should be improved as well.

Originated from https://github.com/sympy/sympy/pull/16381#issuecomment-478355173.

**Comment by czgdp1807** [comment]
@sidhantnagpal I have made the updates in #16381

**Comment by asmeurer** [comment]
If you want this for the release, please get it into `1.4` by tomorrow.

**Comment by sylee957** [comment]
I think it is defined differently in wolfram, than wikipedia
http://mathworld.wolfram.com/GumbelDistribution.html
Can anyone check the sources?

**Comment by sidhantnagpal** [comment]
I believe the difference is in the extreme value distribution for minimum or maximum. In any case, th current behaviour does not follow either.

Reference: https://www.itl.nist.gov/div898/handbook/eda/section3/eda366g.htm

**Comment by czgdp1807** [comment]
[This](http://shodhganga.inflibnet.ac.in/bitstream/10603/25781/11/11_chapter%25203.pdf&ved=2ahUKEwjxjN6o-q3hAhULLo8KHfHOCgUQFjADegQIAxAB&usg=AOvVaw3KWY19RE3iHBLp9zYBzhxY&cshid=1554089531553) suggests that there are two distributions, namely,
1. Gumbel Distribution for minimum
2. Gumbel Distribution for maximum
I think both should be implemented in sympy, or atleast, there should be a provision to pass the parameter for `minimum` and `maximum` in the distribution. 
Waiting for a response.

**Comment by czgdp1807** [comment]
Taking into account the comments of @sylee957 and @sidhantnagpal and [this](https://prism.ucalgary.ca/bitstream/handle/11023/796/ucalgary_2013_ji_chaoqun.pdf?sequence=4&isAllowed=y) document from University Of Calgary, I have the following suggestions,

1. The current Gumbel distribution is not having incorrect pdf, it's just the for the minimum. So, `Wrong Result` label should be removed.
2. The `1.4` branch can go without the fixes in #16523, since, currently no wrong result is there in that branch. Moreover, I think #16523 can be safely closed.
3. The current `crv_types` should have two distributions as described in my above comment. However, to keep the API unchanged, we can pass one more optional keyword, namely, `for_min`, which should be by default `True`. If it is set to `False` by the user, which will return the, object of `GumbelDistributionMaximum`. What are your views @sidhantnagpal @sylee957 ?

**Comment by sylee957** [comment]
I'm supportive for having both variants in SymPy.
I see gumbel is some special case of extreme value distribution, but I wonder if there are only minimum and maximum variant, to be called as 'gumbel' on the earth

**Comment by sidhantnagpal** [comment]
> it's just the one for the minimum

The density for (minimum) Gumbel distribution is different from what the current output is. As I mentioned in my _last comment_, current output does not follow either of them and is therefore, incorrect.

**Comment by sidhantnagpal** [comment]
As far as API is concerned, I think using `Gumbel` is appropriate (and it also does not require any backward compatibility break) while having a configurable keyword like `type='max'` as the default.
The choice of maximum being the default has to do with how the extreme value distributions were theorised. More information at https://en.wikipedia.org/wiki/Generalized_extreme_value_distribution#Link_to_Fr%C3%A9chet,_Weibull_and_Gumbel_families.

**Comment by czgdp1807** [comment]
I have created #16528 for 1.5. The future work section of the first comment mentions the purpose of the PR.
I have also created #16529 for merging into 1.4. It is just for the release and further enhancement will be made in #16528 .

**Comment by sidhantnagpal** [comment]
Fixed in #16529.
